### ISSUE TITLE
Oracle data provider authentication

### ISSUE THREAD

**Issue by wburt** [qgis/QGIS#56535]
### What is the bug or the crash?

For users who do not use the QGIS Authentication System, the QGIS database authentication is dialog opened repeatedly for a single data source. 

### Steps to reproduce the issue

Open QGIS using a profile where oracle credentials are not stored in the Authentication System.
Add a dataset from the oracle data source
Enter your credentials into the database authentication dialog: For .qlr  files this seems to be twice; once when the file is read and once just before data is drawn.
Pan and zoom -- 'Enter Credentials' dialog will reappear. Credentials are accepted but if you pan and zoom the dialog will reappear.

### Versions

I have tested and confirmed this behavior is present for 3.34.4 and 3.32.3 for multiple users. **I do not see this bug in 3.22.7**

QGIS version
3.34.3-Prizren
QGIS code revision
47373234ac
Qt version
5.15.3
Python version
3.9.18
GDAL/OGR version
3.8.3
PROJ version
9.3.1
EPSG Registry database version
v10.098 (2023-11-24)
GEOS version
3.12.1-CAPI-1.18.1
SQLite version
3.41.1
PDAL version
2.6.0
PostgreSQL client version
15.2
SpatiaLite version
5.1.0
QWT version
6.1.6
QScintilla2 version
2.13.4
OS version
Windows 10 Version 2009

Active Python plugins
db_manager
0.1.20
grassprovider
2.12.99
MetaSearch
0.3.6
processing
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

**Comment by troopa81** [comment]
@wburt I failed to reproduce the issue with a sample data of mine, with and without a qlr file, with 3.28 and master.

3.22 is EOL (End of Life), could you please test with the current LTR 3.34.5 ?

> Data can be found -->[here](https://catalogue.data.gov.bc.ca/dataset/natural-resource-nr-areas)

Nothing happen if I click on the link *Access/Download* for `View Generalized BC Geographic Warehouse details (export is not enabled) ` dataset

**Comment by wburt** [comment]
Thank you so much for testing. Same behavior exists in 3.36.0 and 3.34.4. I don't expect this error is data specific as I have tested it with many layers across the few different Oracle database configurations I have access to. The data can be downloaded by clicking the view link from BC Geographic Warehouse Custom Download and then the access/download link in the top right.

**Comment by wburt** [comment]
I still am working to find the source of this issue. I can confirm that it exists for us in 3.36.1. Is there any way of logging to determine if the authentication request is being driven by QGIS or Oracle?

**Comment by troopa81** [comment]
> I still am working to find the source of this issue. I can confirm that it exists for us in 3.36.1. Is there any way of logging to determine if the authentication request is being driven by QGIS or Oracle?

Don't bother, I managed to reproduce it from time to time. I have still to  investigate deeper to understand why and how to fix. I'll take care of it.

**Comment by HeatherHillers** [comment]
Hi, just to let you know, I am seeing the same problem.

This week, I updated my QGIS from QGIS LTR 3.28 to QGIS LTR 3.34.  I kept my user profile.

I loaded a single Oracle Layer from the Add Layer Menu.  I did not enter and auth configuration, or a user, or a password.  I would expect, from previous versions, that QGIS would give me a login when the layer loads, and then remember that login for the rest of my user session.  It would also use that login for any successive oracle layers loaded during the session.

However,  I was getting repeated login screens, same as described by the user above.  At least.  Often it would give me much more than 3 login screens.  I am in Home Office, so the network connection might have been a factor.  But again, in previous versions, I might expect a layer to be a bit slow if the network doesnt behave, but I would not get a repeated login window.

I saw this in layers from multiple databases, also layers loaded from pyqgis.  I did not find a layer that was not misbehaving.  The problem is very consistent.

Once I loaded the layer  with a Authentification from the Authentification Configuration, the problem went away.

Users in my org often trade qlrs or make base projects to load Oracle Layers, so it is quite common to load without a defined Authentification, and let the user log in on layer load.


The error is an ORA 01017 invalid username/password, even though the correct username and password are consistently given.  It makes me wonder if null credentials are being passed by the authentification manager. 

An example qlr of a single layer is attached below, with host, database name, table name and schema name swapped out.
[foo.qlr.txt](https://github.com/qgis/QGIS/files/15062598/foo.qlr.txt)


With the qgis-ltr from last week:


QGIS-Version
3.34.5-Prizren
QGIS-Codeversion
4b308492
Qt-Version
5.15.13
Python-Version
3.12.3
GDAL-Version
3.8.5
PROJ-Version
9.4.0
EPSG-Registraturdatenbankversion
v11.004 (2024-02-24)
GEOS-Version
3.12.1-CAPI-1.18.1
SQLite-Version
3.45.1
PDAL-Version
2.6.3
PostgreSQL-Client-Version
16.2
SpatiaLite-Version
5.1.0
QWT-Version
6.2.0
QScintilla2-Version
2.14.1
BS-Version
Windows 10 Version 2009
